agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AzureMonitorLogProcessor.java

```diff
@@ -27,4 +41,66 @@ public void onEmit(Context context, ReadWriteLogRecord logRecord) {
       logRecord.setAttribute(AiSemanticAttributes.ITEM_COUNT, itemCount);
     }
   }
+
+  @Nullable
+  private static Class<?> getSdkReadWriteLogRecordClass() {
+    try {
+      return Class.forName("io.opentelemetry.sdk.logs.SdkReadWriteLogRecord");
+    } catch (ClassNotFoundException e) {
+      return null;
+    }
+  }
+
+  @Nullable
+  private static Field getLockField() {
+    if (sdkReadWriteLogRecordClass == null) {
+      return null;
+    }
+    try {
+      Field lockField = sdkReadWriteLogRecordClass.getDeclaredField("lock");
+      lockField.setAccessible(true);
+      return lockField;
+    } catch (NoSuchFieldException e) {
+      return null;
+    }
+  }
+
+  @Nullable
+  private static Field getAttributesMapField() {
+    if (sdkReadWriteLogRecordClass == null) {
+      return null;
+    }
+    try {
+      Field attributesMapField = sdkReadWriteLogRecordClass.getDeclaredField("attributes");
+      attributesMapField.setAccessible(true);
+      return attributesMapField;
+    } catch (NoSuchFieldException e) {
+      return null;
+    }
+  }
+
+  private static void setAttributeExceptionLogged(Span span, ReadWriteLogRecord logRecord) {
+    if (lockField == null || attributesMapField == null) {
+      return;
+    }
+    String stacktrace = null;
+    try {
+      synchronized (lockField) {
+        stacktrace =
+            ((AttributesMap) attributesMapField.get(logRecord))
```

Comment:
I'd suggest null checking the `AttributesMap` for safety in case it gets optimized (lazy created) in the future